### PR TITLE
Deprecate `conda_build.api.get_output_file_path`

### PR DESCRIPTION
**Pull request by ryanskeith**
### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by beeankha** [comment]
pre-commit.ci autofix

**Comment by ryanskeith** [comment]
Do we feel okay with doing this deprecation without using the @deprecated function?  @beeankha @kenodegard  I feel we are ok as users have been warned for the last 7 years, but would be perfectly fine backtracking.  The only internal use we have is for the tests.  Everywhere else has been using the newer name.

**Comment by ryanskeith** [comment]
Will do.

File: conda_build/api.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 from os.path import dirname, expanduser, join
 from pathlib import Path
 
+from conda.deprecations import deprecated
+
 # make the Config class available in the api namespace
 from .config import DEFAULT_PREFIX_LENGTH as _prefix_length
 from .config import Config, get_channel_urls, get_or_merge_config
```
Comment (kenodegard) [inline conda_build/api.py:24]: The deprecations module is also available in conda-build:

```suggestion
# make the Config class available in the api namespace
from .config import DEFAULT_PREFIX_LENGTH as _prefix_length
from .config import Config, get_channel_urls, get_or_merge_config
from .deprecations import deprecated
```

File: news/5208-deprecate-get_output_file_path [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* Mark `conda_build.api.get_output_file_path` as deprecated. (#5208)
```
Comment (kenodegard) [inline news/5208-deprecate-get_output_file_path:11]:
```suggestion
* Mark `conda_build.api.get_output_file_path` as deprecated. Use `conda_build.api.get_output_file_paths` instead. (#5208)
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

**Review by kenodegard — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**